### PR TITLE
Add multiple merge support, tests & refactor example to use new multiple merge

### PR DESCRIPTION
**Pull request by kyl191**
Removing lists is still not supported. However:
- Dicts can be merged with an arbitrary (>1) number of dicts, instead of just 2
- Some comments improved
- Example EC2 script reworked to use more of the new functionality

Future extensions: 
- automated 2to3 for Python 3 support in install
- rework the merge indicator (I suspect it can be reduced to a single boolean flag, getting rid of target & other references)

### CODE REVIEW

File: yamlstratus/yamlstratus.py [hunk 1]
```diff
@@ -207,45 +207,39 @@ def join_text(self, node):
...
-                           + "'startingFrom' and 'mergeWith'")
-
-        if isinstance(mappings['startingFrom'], list):
-            if not isinstance(mappings['mergeWith'], list):
+        # deep is undocumented, but it forces depth-first traversal
+        mappings = self.construct_mapping(node, deep=True)
+        if len(mappings) < 2:
+            raise KeyError("!merge extension requires at least 2 child nodes ")
+
+        # Mapping should be in order of appearance. But because we use keys,
```
Comment (usmanismail) [inline yamlstratus/yamlstratus.py]: Order of appearance seems to be irrelevant, seems to use key name for order.

**Comment by usmanismail** [comment]
@kyl191 Minor comment but looks good. Passing to @shannonwhite

File: example/ec2/Files.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+"/etc/cfn/hooks.d/cfn-auto-reloader.conf":
+    content: !jtext |
+              [cfn-auto-reloader-hook]
+              triggers=post.update
+              path=Resources.WebServerInstance.Metadata.AWS::CloudFormation::Init
+              action=/opt/aws/bin/cfn-init -v --stack !{Ref: "AWS::StackName"!} --resource WebServerInstance --configsets InstallAndRun --region !{Ref: "AWS::Region"!}
+              runas=root
+    mode: "000400"
+    owner: root
+    group: root
```
Comment (shannonwhite) [inline example/ec2/Files.yaml]: Missing newline

File: test/multi_merge_extension.json [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+[
+    "f",
+    "h",
+    "c",
+    "a",
+    "b",
+    "d"
+]
+
+]
```
Comment (shannonwhite) [inline test/multi_merge_extension.json]: Missing newline

**Comment by shannonwhite** [comment]
* Need to get the README changes in this PR, too.
* The merge order needs clarification. It cannot be arbitrary

File: test/test_yamlstratus.py [hunk 4]
```diff
@@ -197,6 +197,16 @@ def test_merge_syntax_errors():
...
                             - value2
+                """
+    for yaml in yaml_str.split('---')[1:]:
+        try:
+            yamlstratus.load_as_json(yaml, include_dirs=["test"])
+            assert False
+        except KeyError:
+            pass
+
+def test_merge_new_keys():
```
Comment (shannonwhite) [inline test/test_yamlstratus.py]: Just eliminate this test: not sure it tests anything valuable anymore.

Comment (kyl191) [inline test/test_yamlstratus.py]: It's testing merge nodes with a single hierarchy. I've renamed the test to make it more explicit.

**Comment by usmanismail** [comment]
@shannonwhite @kyl191 Whats remaining here?
